From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Thu, 5 Mar 2015 10:33:25 +0100
Subject: [PATCH] missing: define __NR_kcmp if necessary

Needed for Kernel Headers < v3.5

Not for upstream, but needed for host-systemd.

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 src/basic/missing.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/basic/missing.h b/src/basic/missing.h
index 461c122ca9c3..5b3093d8a3f6 100644
--- a/src/basic/missing.h
+++ b/src/basic/missing.h
@@ -1072,6 +1072,11 @@ static inline int renameat2(int oldfd, const char *oldname, int newfd, const cha
 #define RENAME_NOREPLACE (1 << 0)
 #endif
 
+#ifndef __NR_kcmp
+# warning "__NR_kcmp unknown"
+# define __NR_kcmp 0xffffffff
+#endif
+
 #if !HAVE_DECL_KCMP
 static inline int kcmp(pid_t pid1, pid_t pid2, int type, unsigned long idx1, unsigned long idx2) {
 #if defined(__NR_kcmp)
